lib/nixd/src/nix/Parser.y

```diff
@@ -0,0 +1,691 @@
+// \$\$ = (.*);
+// { $$$$ = $1;  data->locations[$$$$] = CUR_POS; }
+// d|ata->symbols.create\(([^ )]*)\)
+// [&] () { auto Sym = [&] () { auto Sym = data->symbols.create($1); data->symbolLocations.insert({Sym, CUR_POS}); return Sym; }(); data->symbolLocations.insert({Sym, CUR_POS}); return }()
```

Comment:
Might be something wrong with this change. Because symbols are "uniquely allocated".